Assets/__Scripts/BusinessLogic/CubeCreation/UnitManager.cs

```diff
@@ -46,16 +48,25 @@ public void DidSelectUnitAndArrow(UnitController unit, GameObject creationArrow)
     {
         Vector3 distance = (unit.representation.transform.position - creationArrow.transform.position) * 2;
         Vector3 newUnitPosition = unit.transform.position - distance;
-        if (UnitCounter.count == UnitCounter.MAX_NUMBER_OF_CUBES)
+
+        int number;
+        if (Int32.TryParse(numberOfCubes.text, out number))
         {
-#if UNITY_EDITOR
-            EditorUtility.DisplayDialog("Cube limit reached", "Cannot add more cubes! " +
-                "The maximum number of cubes permitted is 5000.", "Ok");
-#endif
-        }
-        else
+            while (number != 0)
+            {
+                CreateUnit(newUnitPosition);
+                newUnitPosition -= distance;
+                number--;
+
+                if (UnitCounter.count == UnitCounter.MAX_NUMBER_OF_CUBES)
+                {
+                    EditorUtility.DisplayDialog("Cube limit reached", "Cannot add more cubes! " +
+                        "The maximum number of cubes permitted is 5000.", "Ok");
```

Comment:
Change it so it's like 
`"The maximum number of cubes permitted is " + UnitCounter.MAX_NUMBER_OF_CUBES`